python/vast/utils/asyncio.py

```diff
@@ -0,0 +1,44 @@
+import asyncio
+from signal import SIGINT, SIGTERM, SIGHUP
+
+import vast.utils.logging
+
+logger = vast.utils.logging.get()
+
+
+async def shutdown(loop, signal=None):
+    """Cleanup tasks tied to the service's shutdown."""
+    if signal:
+        logger.info(f"received exit signal {signal.name}")
+    tasks = [t for t in asyncio.all_tasks() if t is not asyncio.current_task()]
+    [task.cancel() for task in tasks]
+    logger.debug(f"cancelling {len(tasks)} outstanding tasks")
+    await asyncio.gather(*tasks, return_exceptions=True)
+    loop.stop()
+
+
+def handle_exception(loop, context):
+    # context["message"] will always be there, but context["exception"] may not
```

Comment:
Why? Where is the context object here coming from?

Reply:
This is passed in here:
```python
    loop.set_exception_handler(handle_exception)
```

The `context` is part of the expected signature.